evm/src/cpu/stack.rs

```diff
@@ -0,0 +1,387 @@
+use itertools::izip;
+use plonky2::field::extension::Extendable;
+use plonky2::field::packed::PackedField;
+use plonky2::field::types::Field;
+use plonky2::hash::hash_types::RichField;
+use plonky2::iop::ext_target::ExtensionTarget;
+
+use crate::constraint_consumer::{ConstraintConsumer, RecursiveConstraintConsumer};
+use crate::cpu::columns::ops::OpsColumnsView;
+use crate::cpu::columns::CpuColumnsView;
+use crate::cpu::membus::NUM_GP_CHANNELS;
+use crate::memory::segments::Segment;
+
+#[derive(Clone, Copy)]
+struct StackBehavior {
+    num_pops: usize,
+    pushes: bool,
+    disable_other_channels: bool,
+}
+
+const STACK_BEHAVIORS: OpsColumnsView<Option<StackBehavior>> = OpsColumnsView {
+    stop: None, // TODO
+    add: Some(StackBehavior {
```

Comment:
I think it would be nice to have a few shared consts like

```rust
const BINARY_OP: Option<StackBehavior> = Some(StackBehavior {
    num_pops: 2,
    pushes: true,
    disable_other_channels: true,
});
```

to make the `STACK_BEHAVIORS` definition more succinct.